lib/dal/src/attribute/value.rs

```diff
@@ -1151,6 +1240,20 @@ impl AttributeValue {
             AttributeValueError::AttributePrototypeNotFound(self.id, *ctx.visibility())
         })?;
 
+        // Check if the function is one of the "si:set*", or "si:unset" functions, as these are
+        // special, and can't actually be re-run. Their values are static anyway, so re-running it
+        // wouldn't change anything. The "si:setObject", "si:setArray", and "si:setMap" functions
+        // are a bit special, however, as the "local" value will always be an empty object, array,
+        // or map.
+        let func = Func::get_by_id(ctx, &attribute_prototype.func_id())
+            .await?
+            .ok_or_else(|| {
+                AttributeValueError::MissingFunc(format!("Unable to get func for {:?}", self.id()))
+            })?;
+        if func.name().starts_with("si:set") || func.name() == "si:unset" {
```

Comment:
it might be safer to use the `InstrincFunc::maybe_from_str(func.name())` and just see that it's `Some` (but not `Some(IntrinsicFunc::Identity)`, just in case someone adds a custom function that starts with si:set